### PR TITLE
Add tests for API method tcms.bugs.api.filter()

### PR DESCRIPTION
**Pull request by mfonism**
This PR adds tests for the `filter()` api method in `tcms/bugs/api.py` as per #1597

### CODE REVIEW

File: tcms/bugs/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from tcms.bugs.tests.factory import BugFactory
+from tcms.rpc.tests.utils import APITestCase
+from tcms.tests.factories import UserFactory
+
+
+class TestFilter(APITestCase):
+    """Test Bug.filter"""
+
+    def setUp(self):
+        super(TestFilter, self).setUp()
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:13]: Use `super().setUp()` - Python 3 version of super.

File: tcms/bugs/tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        atodorov = UserFactory(username='atodorov')
+        asankov = UserFactory(username='asankov')
+        BugFactory(status=False, assignee=asankov)
+        BugFactory(status=True, assignee=asankov)
+        BugFactory(status=True, assignee=atodorov)
+
+    def test_filter_bugs(self):
+        result = self.rpc_client.Bug.filter({'status': 'True'})
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:22]: Try `True` as a boolean, not a string. It should work.

File: tcms/bugs/tests/test_api.py [hunk 3]
```diff
@@ -93,3 +93,27 @@ def verify_api_with_permission(self):
...
+        self.yet_another_bug = BugFactory(status=True)
+
+    def test_filter(self):
+        result = self.rpc_client.Bug.filter({"status": True})
+        self.assertIsInstance(result, list)
+        self.assertEqual(len(result), 2)
+        # sort by pk of contained dicts
+        result.sort(key=lambda d: d["pk"])
+        self.assertEqual(result[0]["pk"], self.another_bug.pk)
+        self.assertEqual(result[1]["pk"], self.yet_another_bug.pk)
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:115]: The order is not guaranteed b/c the underlying query in .filter() doesn't sort the results. Use assertIn/assertNotIn to check the results or iterate over the resulting list to check things.

**Comment by mfonism** [comment]
@atodorov I've implemented the changes you requested.

**Pull request merged**